### DISCUSSION TITLE
Allowing GET routes to have input arguments coming indistinctively from Query OR Body

### DISCUSSION THREAD

**Discussion by farbiondriven-spire** [fastapi/fastapi#10981] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, FastAPI, Query
from pydantic import BaseModel

app = FastAPI()

class InputData(BaseModel):
    name: str = Query(..., examples=["A Fancy Name"])
    non_mandatory_data: str | None = Query(None)


@app.get("/")
async def items(params: InputData = Depends()):
    return params
# Only Query will be accepted and inputs in JSON won't be loaded

@app.get("/")
async def items(params: InputData):
    return params
# The other way around
```

### Description

I converted an existing server from flask/flask-restx to FastAPI, when deploying the new version I 
faced an issue with some existing clients calling several GET endpoints putting their input parameters in the Body.

I put in place a DIRTY workaround for the moment, but I would like to know if there's a cleaner approach/any intention to put a feature to make the API more forgiving as in flask_restx.

```
async def allow_json_parameters(request: Request):
   # somewhere else I check if there's
    if not request.query_params:
        json = await request.json()
        request._query_params = QueryParams(json)  # pylint: disable=protected-access

# then on the route
@router.get("/", dependencies=[Depends(allow_json_parameters)])
```

### Operating System

Linux

### FastAPI Version

0.103.2

### Pydantic Version

2.5.3

### Python Version

3.11.1

**Comment by YuriiMotov** [comment, marked as answer]
AFAIK, you can't clearly define this in openapi schema.
And I think you shouldn't go this way. I would suggest creating two endpoints each accept parameters it's own way. This way, clients can chose the endpoint depending on how they are going to pass data.
